glusterd2/events/eventhandler.go

```diff
@@ -154,3 +162,69 @@ func (h *handler) Handle(e *api.Event) {
 func (h *handler) Events() []string {
 	return h.events
 }
+
+func getJWTToken(eventtype, secret string) string {
+	claims := &jwt.StandardClaims{
+		IssuedAt:  time.Now().Unix(),
+		ExpiresAt: time.Now().Add(time.Second * 10).Unix(),
+		Subject:   eventtype,
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	ss, err := token.SignedString([]byte(secret))
+	if err != nil {
+		return ""
+	}
+	return ss
+}
+
+//WebhookPublish sends event to registered webhooks
+func WebhookPublish(webhook *eventsapi.Webhook, e *api.Event) error {
+	message, err := json.Marshal(e)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"name":  e.Name,
+			"error": err.Error(),
+		}).Error("failed to marshal event")
+		return err
+	}
+
+	body := strings.NewReader(string(message))
+
+	req, err := http.NewRequest("POST", webhook.URL, body)
+	if err != nil {
+		log.WithError(err).Error("error forming the request object")
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	if webhook.Token != "" {
+		req.Header.Set("Authorization", "bearer "+webhook.Token)
+	}
+
+	if webhook.Secret != "" {
+		token := getJWTToken(e.Name, webhook.Secret)
+		req.Header.Set("Authorization", "bearer "+token)
+	}
+
+	tr := &http.Transport{
+		DisableCompression:    true,
+		DisableKeepAlives:     true,
```

Comment:
Are these good defaults? It seems odd to me to see these hard-coded here, although I'll admit to not entirely understanding this subsystem.
At the very least I think comments explaining why these settings were chosen would be helpful.